### PR TITLE
[CA-1494] Prevent Workspace Creation in Non-Ready Billing Projects

### PR DESCRIPTION
**Pull request by ehigham**
Suggestions for a good error message for this welcome.

- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by ehigham** [comment]
Seems to be failing post merge 
```
[info] *** 1 SUITE ABORTED ***
[error] Error during tests:
[error] 	org.broadinstitute.dsde.rawls.entities.datarepo.DataRepoBigQuerySanitizationSpec
[error] (rawlsCore / Test / test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 1283 s (21:23), completed Sep 14, 2021, 8:43:54 PM
Build step 'Conditional steps (multiple)' marked build as failure
```

**Review by s-rubenstein — APPROVED** [review]
LGTM

File: core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala [hunk 1]
```diff
@@ -1282,20 +1282,19 @@ class WorkspaceServiceSpec extends AnyFlatSpec with ScalatestRouteTest with Matc
     )
   }
 
-  // TODO: This test will need to be deleted when implementing https://broadworkbench.atlassian.net/browse/CA-947
```
Comment (marctalbott) [inline core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala:1285]: This TODO is still relevant I believe -- once we remove the Terra billing project creation status entirely, we won't be able to return a 400 if it's not Ready

**Review by marctalbott — APPROVED** [review]
Approved assuming TODO is re-added

File: core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala (hunk 1 above)
Comment (ehigham) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala:1285]: Is there another way to tell if a v1 billing project failed to be created?

Comment (ehigham) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala:1285]: For posterity, CA-947 involves deleting all errored billing projects, therebye making this change obsolete on completion.

Comment (marctalbott) [reply core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceServiceSpec.scala:1285]: There isn't another way to tell that comes to mind. It is possible that we will decide not to to CA-947 at all in favor of keeping the creation status (even if it's a bit obsolete for v2 projects), but I'm not sure. We can remove this TODO if we close out CA-947

**Pull request merged**